lib/schema/SingleNestedPath.js

```diff
@@ -38,7 +38,7 @@ function SingleNestedPath(schema, path, options) {
   this.caster = _createConstructor(schema);
   this.caster.path = path;
   this.caster.prototype.$basePath = path;
-  this.schema = schema;
```

Comment:
This change is also unnecessary. SchemaTypes can keep `this.schema`, so stuff under `lib/schema` doesn't need to change. Files under `lib/types` are the ones that need to change.